azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/resources/templates.json

```diff
@@ -600,6 +600,214 @@
       "files": {
         "function.java": "package $packageName$;\r\n\r\nimport com.fasterxml.jackson.core.JsonProcessingException;\r\nimport com.fasterxml.jackson.databind.JsonNode;\r\nimport com.fasterxml.jackson.databind.ObjectMapper;\r\nimport com.microsoft.azure.functions.ExecutionContext;\r\nimport com.microsoft.azure.functions.OutputBinding;\r\nimport com.microsoft.azure.functions.annotation.FunctionName;\r\nimport com.microsoft.azure.functions.dapr.annotation.DaprStateOutput;\r\nimport com.microsoft.azure.functions.dapr.annotation.DaprTopicTrigger;\r\n\r\n\r\npublic class $className$ {\r\n\r\n    \/*\r\n    Visit https:\/\/aka.ms\/azure-functions-dapr to learn how to use the Dapr extension.\r\n    These tasks should be completed prior to running :\r\n      1. Install Dapr\r\n      2. Change the bundle name in host.json to \"Microsoft.Azure.Functions.ExtensionBundle.Preview\" and the version to \"[4.*, 5.0.0)\"\r\n    Start function app with Dapr: dapr run --app-id functionapp --app-port 3001 --dapr-http-port 3501 -- mvn azure-functions:run\r\n    Invoke function app in cmd promt: dapr publish --pubsub pubsub --publish-app-id functionapp --topic Q --data '{\\\"value\\\": { \\\"orderId\\\": \\\"42\\\" } }'\r\n    *\/\r\n    @FunctionName(\"DaprTopicTriggerFunc\")\r\n    public String run(\r\n        @DaprTopicTrigger(\r\n            pubSubName = \"pubsub\",\r\n            topic = \"Q\")\r\n            String request,\r\n        @DaprStateOutput(\r\n            stateStore = \"statestore\",\r\n            key = \"product\")\r\n        OutputBinding<String> product,\r\n        final ExecutionContext context) throws JsonProcessingException {\r\n        context.getLogger().info(\"Java DaprTopic trigger with DaprState output binding function processed a request from the Dapr Runtime.\");\r\n\r\n        \/\/ Get the CloudEvent data from the request body as a JSON string\r\n        ObjectMapper objectMapper = new ObjectMapper();\r\n        JsonNode jsonNode = objectMapper.readTree(request).get(\"data\");\r\n        String serializedJson = objectMapper.writeValueAsString(jsonNode);\r\n        product.setValue(serializedJson);\r\n\r\n        return serializedJson;\r\n    }\r\n}"
       }
+    },
+    {
+      "id": "AssistantSkillTrigger-Java",
+      "metadata": {
+        "name": "AssistantSkillTrigger",
+        "description": "$AssistantSkillTrigger_description",
+        "defaultFunctionName": "assistantSkillTriggerJava",
+        "language": "Java"
```

Comment:
for properties you want users to config during `mvn azure-functions:add`, please use property userPrompt to config them, here is an example
```
...
"language": "Java",
"userPrompt": [
          "name",
          "functionDescription"
           ...
],
...
```